src/appmixer/mongodb/common.js

```diff
@@ -26,7 +26,23 @@ module.exports = {
             options.tlsAllowInvalidCertificates = true;
         }
         const client = new MongoClient(context.connectionUri, context.tlsCAFileContent && options);
-        await client.connect();
+        try {
+            await client.connect();
+        } catch (error) {
+            if (context.tlsCAFileContent) {
+                // Removing the temporary file and directory if the connection fails
+                await new Promise((resolve, reject) => {
```

Comment:
perhaps it could be simplified as: 

` await fs.rm(path, { recursive: true, force: true });` 

using the `const fs = require('fs').promises;`

Reply:
I wasn't able to make it work with `fs.promises` - the folder always remained even though the file was removed.

Instead I refactored the code to extract the function.